docs/reference/modules/ROOT/pages/http.adoc

```diff
@@ -73,183 +73,276 @@ If the node was started on `localhost:3000`, you can connect to it by doing the
-|===
+All of the REST endpoints return both `application/edn` and `application/transit+json` (see transit https://github.com/cognitect/transit-clj[here]) - individual endpoints may return additional types - see the docs below. To set up your node to with the REST API, see the xref:http.adoc[HTTP Feature Module].
 
-[#entity]
-=== GET `/entity/[:key]`
+[#status]
+=== GET `/_crux/status`
+
+Returns the current status information of the node.
+
+[source,bash]
+----
+curl -X GET \
+     -H "Content-Type: application/edn" \
+     $CRUX_URL/_crux/status
+----
 
-Takes a key and, optionally, a `:valid-time` and/or `:transact-time` (defaulting to now). Returns the value stored under that key at those times.
+[#entity]
+=== GET `/_crux/entity`
 
-See xref:bitemporality.adoc[Bitemporality] for more information.
+Returns the document map for a particular entity.
 
 [source,bash]
 ----
 curl -X GET \
      -H "Content-Type: application/edn" \
-     $nodeURL/entity/:tommy
+     $CRUX_URL/_crux/entity?eid=:tommy
 ----
 
-[source,clj]
+==== Query Parameters
+
+.*Required Parameters*
+* `eid` (Crux ID)
+
+.*Optional Parameters*
+* `valid-time` (date, defaulting to now)
+* `transaction-time` (date, defaulting to latest transaction time)
+
+=== GET `/_crux/entity?history=true`
+
+Returns the history of a particular entity.
+
+[source,bash]
 ----
-{:crux.db/id :tommy, :name "Tommy", :last-name "Petrov"}
+curl -X GET \
+     -H "Content-Type: application/edn" \
+     $CRUX_URL/_crux/entity?eid=:tommy&history=true&sort-order=asc
 ----
 
+.*Required Parameters*
+* `eid` (Crux ID)
+* `sort-order` (_required_, either `asc` or `desc`)
+
+.*Optional Parameters*
+* `with-corrections` (boolean, default false): includes bitemporal corrections in the response, inline, orted by valid-time then transaction-time
+* `with-docs` (boolean, default false): includes the documents in the response sequence, under the `:crux.db/doc` key
+* `start-valid-time`, `start-transaction-time` (inclusive, default unbounded): bitemporal co-ordinates to start at
+* `end-valid-time`, `end-transaction-time` (exclusive, default unbounded): bitemporal co-ordinates to stop at
+
+[#entity-tx]
+=== GET `/_crux/entity-tx`
+
+Returns the transaction details for an entity - returns a map containing the tx-id and tx-time.
+
 [source,bash]
 ----
 curl -X GET \
      -H "Content-Type: application/edn" \
-     $nodeURL/entity/:tommy?valid-time=1999-01-08T14%3A03%3A27%3A254-00%3A00
+     $CRUX_URL/_crux/entity-tx?eid=:tommy
 ----
 
-[source,clj]
+==== Query Parameters
+
+.*Required Parameters*
+* `eid` (Crux ID)
+
+.*Optional Parameters*
+* `valid-time` (date, defaulting to now)
+* `transaction-time` (date, defaulting to latest transaction time)
+
+[#query]
+=== GET `/_crux/query`
+
+Takes a datalog query and returns its results. Results are also available in `text/csv` and `text/tsv` formats (can force negotiation of these by using the `/_crux/query.csv` and `/_crux/query.tsv` endpoints respectively).
```

Comment:
I wonder whether we should extend this both to `.edn`, and to the other endpoints, as a general pattern - wdyt?

(Maybe not one for this PR)

Reply:
Perhaps - though it does wind up making the handler pretty messy and you wind up repeating a lot of things - perhaps if there were a cleaner way of doing it?

Reply:
As discussed - not on this PR